### PR TITLE
NTBS-2944: Add notifying service and 12/24/36 month responsible clinic

### PR DESCRIPTION
**Pull request by LeoCie**
I also fixed a bug where notifications with no diagnosis made/treatment start event were not having outcomes generated in reporting (this was happening with post mortem cases as some only have death events)

### CODE REVIEW

File: source/dbo/Stored Procedures/Power BI Reporting/uspGenerateRecordOutcome.sql [hunk 1]
```diff
@@ -25,13 +25,13 @@ BEGIN TRY
 	FROM
 		[dbo].[Outcome] o
 		INNER JOIN [$(NTBS)].[dbo].[Notification] n ON n.NotificationId = o.NotificationId
-		INNER JOIN [$(NTBS)].[dbo].[TreatmentEvent] te ON te.NotificationId = o.NotificationId
-	WHERE te.TreatmentEventType = 'TreatmentStart' OR te.TreatmentEventType = 'DiagnosisMade'
+		LEFT JOIN [$(NTBS)].[dbo].[TreatmentEvent] te ON te.NotificationId = o.NotificationId
+			AND (te.TreatmentEventType = 'TreatmentStart' OR te.TreatmentEventType = 'DiagnosisMade')
```
Comment (LeoCie) [inline source/dbo/Stored Procedures/Power BI Reporting/uspGenerateRecordOutcome.sql:29]: This was the bug fix

File: source/dbo/Stored Procedures/Power BI Reporting/uspGenerateNtbsCaseRecord.sql [hunk 2]
```diff
@@ -32,6 +32,16 @@ BEGIN TRY
...
 
+	WITH firstTransferOut AS
+	(SELECT t.NotificationId, TbServiceCode, tbs.[Name]
+		FROM (SELECT DISTINCT NotificationId FROM [$(NTBS)].dbo.TreatmentEvent) te
+		CROSS APPLY (SELECT TOP 1 *
+			FROM [$(NTBS)].dbo.TreatmentEvent te2
+			WHERE te2.NotificationId = te.NotificationId AND TreatmentEventType = 'TransferOut'
+			ORDER BY EventDate) t
+		JOIN [$(NTBS)].ReferenceData.TbService tbs ON tbs.Code = t.TbServiceCode)
+	SELECT * INTO #TempTransfers FROM firstTransferOut;
```
Comment (JoshCadney) [inline source/dbo/Stored Procedures/Power BI Reporting/uspGenerateNtbsCaseRecord.sql:43]: The "notifying" fields are really just "TB service responsible at 0 months" so I think we should calculate them all in the same file, and use similar logic for all

**Review by JoshCadney — CHANGES_REQUESTED** [review]

File: source/dbo/Stored Procedures/Power BI Reporting/uspGenerateRecordOutcome.sql [hunk 3]
```diff
@@ -25,13 +25,13 @@ BEGIN TRY
...
 		
 	--Call the stored proc for each outcome period: 1, 2 and 3 (12 month, 24 month, 36 month)
 	EXEC [dbo].[uspGenerateReusableOutcomePeriodic] 1
 	EXEC [dbo].[uspGenerateReusableOutcomePeriodic] 2
-	EXEC [dbo].[uspGenerateReusableOutcomePeriodic] 3
+	EXEC [dbo].[uspGenerateReusableOutcomePeriodic] 3;
```
[the remark this answers is not in the document]
Comment (LeoCie) [reply source/dbo/Stored Procedures/Power BI Reporting/uspGenerateRecordOutcome.sql:34]: I'm fairly surely it's actually needed here to terminate the exec statement before the "WITH".
Without it this happens:
image

[the remark this answers is not in the document]
Comment (JoshCadney) [reply source/dbo/Stored Procedures/Power BI Reporting/uspGenerateRecordOutcome.sql:34]: Very strange! In that case, could you add a semi colon to each of those exec statement, so it's less jarring to the eye!

**Review by JoshCadney — APPROVED** [review]
Looks good :-)

**Pull request merged**